### PR TITLE
feat(user data validation): Validate user input during registration

### PR DESCRIPTION
**Pull request by nyabongoedgar**
**What does this PR do ?**
* Adds functionality that validates user input on registration.

**Description of the tasks to be completed?**

    Sends a descriptive validation message when 
 - A user inputs a username that does not start with a letter e.g &^%edgar is not a valid username.
    A valid username would be edgar and it should not be less than 4 letters in length.
- A user inputs a password that is less than 8 letters in length and is not alphanumeric or lacks an uppercase letter.
- A user enters an email that is not vaild, e.g at.com, a valid email would be at@gmail.com

**How should this be manually tested?**
* Visit the Sign up route at /api/users/ , try registering a user with an empty body, or a body without a username field, or password or email, try an email that is not valid.

**What are the relevant pivot tracker stores?**
* #164069307

### CODE REVIEW

File: tests/test_validation/test_validation.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
+from rest_framework import status
+from rest_framework.test import APITestCase
+import re
+
+
+class Test_Validation(APITestCase):
```
Comment (PatrickCmd) [inline tests/test_validation/test_validation.py]: @nyabongoedgar following the PEP8 style guidelines we don't write class names with underscores.

File: authors/settings.py [hunk 2]
```diff
@@ -123,6 +123,11 @@
...
 }
-
+EMAIL_BACKEND = 'django.core.mail.backends.smtp.EmailBackend'
+EMAIL_USE_TLS = True
+EMAIL_HOST = os.environ.get('EMAIL_HOST', '')
+EMAIL_PORT = os.environ.get('EMAIL_PORT', '')
+EMAIL_HOST_USER = os.environ.get('EMAIL_HOST_USER', '')
+EMAIL_HOST_PASSWORD = os.environ.get('EMAIL_HOST_PASSWORD', '')
 
 django_heroku.settings(locals())
```
Comment (Stanley-Okwii) [inline authors/settings.py:133]: add new line at the end of the file

File: tests/authentication/test_authentication.py [hunk 3]
```diff
@@ -55,26 +56,70 @@ def test_for_missing_email(self):
...
         response = self.client.post('/api/users/', data, format='json')
 
-        data = {"user": {"email": "akram@gmail.com", "password":
-                "akrammukasa"}}
-        response = self.client.post('/api/users/login/', data, format='json')
+        login_data = {"user": {"email": "akram@gmail.com", "password":
+                               "aKrammukasa1234"}}
+        self.client.post('/api/users/verify/?token=' + response.data['token'])
+        response = self.client.post(
+            '/api/users/login/', login_data, format='json')
```
Comment (Stanley-Okwii) [inline tests/authentication/test_authentication.py:74]: Please create custom methods to handle registration, verification, and log in. instead you calling them through many requests in a single test.

**Review by Stanley-Okwii — APPROVED** [review]

File: tests/test_validation/test_validation.py (hunk 1 above)
Comment (nyabongoedgar) [reply tests/test_validation/test_validation.py]: @PatrickCmd 
Thank you. Class name is now TestValidation(APITestCase).

**Review by PatrickCmd — APPROVED** [review]

**Pull request merged**